app/models/tag.rb

```diff
@@ -17,6 +17,7 @@
 #  published_groups :text(16777215)
 #  child_ordering   :string(255)      default("alphabetical"), not null
 #  index            :integer          default(0), not null
+#  archived         :boolean          default(FALSE), not null
```

Comment:
I'm not convinced this is the best way to model this.  We already have a state column, and a state machine. I think we should add archived as an additional state for this. Otherwise we'd end up with tags that are both published and archived at the same time.